src/ProjectTemplates/Shared/ProcessLock.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Templates.Test.Helpers
+{
+    public class ProcessLock
+    {
+        public static readonly ProcessLock DotNetNewLock = new ProcessLock("dotnet-new");
+        public static readonly ProcessLock NodeLock = new ProcessLock("node");
+
+        public ProcessLock(string name)
+        {
+            Name = name;
+            Semaphore = new SemaphoreSlim(1);
+        }
+
+        public string Name { get; }
+        private SemaphoreSlim Semaphore { get; }
+
+        public async Task WaitAsync(TimeSpan? timeout = null)
+        {
+            timeout ??= TimeSpan.FromMinutes(2);
+            Assert.True(await Semaphore.WaitAsync(timeout.Value), $"Unable to acquire process lock for process {Name}");
+        }
+
+        public void Release()
+        {
+            Semaphore.Release();
+        }
+    }
+}
```

Comment:
Is there a way we can avoid introducing this class? I don't get what value it adds other than throwing an exception. Couldn't we encapsulate that into a helper method instead?

Reply:
How strongly do you feel about this? Seems like more work to undo what is perfectly reasonable code.

Reply:
I want to avoid adding complexity where we don't need it on this code. I want to keep this code as simple and concise as possible. Otherwise it becomes unmanageable. The important thing here is the name and we can probably just need a method for it (unless I'm missing something).

We've been careless in this area before just adding stuff for "fixes" and it ends up causing trouble, so I want to keep things as simple as possible. I don't feel strongly about it, so I'll leave it up to you, but my preference is that we keep things as lean as possible.